services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowConnectedDeviceSpec.groovy

```diff
@@ -396,6 +399,187 @@ srcLldpDevices=#newSrcEnabled, dstLldpDevices=#newDstEnabled"() {
+            it.multiTable = true
+            it.switchLldp = true
+        })
+
+        when: "Device sends an lldp packet into a free port"
+        def lldpData = LldpData.buildRandom()
+        def vlan = 123
+        new ConnectedDevice(traffExamProvider.get(), tg, vlan).withCloseable {
+            it.sendLldp(lldpData)
+        }
+
+        then: "Corresponding device is detected on a switch port"
+        Wrappers.wait(WAIT_OFFSET) {
+            with(northboundV2.getConnectedDevices(sw.dpId).ports) {
+                it.size() == 1
+                it[0].portNumber == tg.switchPort
+                it[0].lldp.first().vlan == vlan
+                verifyEquals(it[0].lldp.first(), lldpData)
+            }
+        }
+
+        cleanup: "Turn off devices prop, remove connected devices"
+        database.removeConnectedDevices(sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialProps)
+    }
+
+    @Ignore("not yet implemented")
+    def "Able to detect device both per flow and per switch at the same time"() {
+        given: "A switch with devices feature turned on"
+        def tg = topology.activeTraffGens[0]
+        def sw = tg.switchConnected
+        def initialProps = northbound.getSwitchProperties(sw.dpId)
+        switchHelper.updateSwitchProperties(sw, northbound.getSwitchProperties(sw.dpId).tap {
+            it.multiTable = true
+            it.switchLldp = true
+        })
+
+        and: "A single-sw flow with lldp device feature 'on'"
+        def flow = flowHelper.randomFlow(sw, sw).tap {
+            it.source.detectConnectedDevices = new DetectConnectedDevicesPayload(true, false)
+        }
+        flowHelper.addFlow(flow)
+
+        when: "Device sends an lldp packet into a flow port"
+        def lldpData = LldpData.buildRandom()
+        new ConnectedDevice(traffExamProvider.get(), tg, flow.source.vlanId).withCloseable {
+            it.sendLldp(lldpData)
+        }
+
+        then: "Corresponding device is detected on a switch port with reference to corresponding flow"
+        Wrappers.wait(WAIT_OFFSET) {
+            with(northboundV2.getConnectedDevices(sw.dpId).ports) {
+                it.size() == 1
+                it[0].portNumber == tg.switchPort
+                it[0].lldp.first().flowId == flow.id
+                it[0].lldp.first().vlan == flow.source.vlanId
+                verifyEquals(it[0].lldp.first(), lldpData)
+            }
+        }
+
+        then: "Device is also visible as a flow device"
+        Wrappers.wait(WAIT_OFFSET) {
+            with(northbound.getFlowConnectedDevices(flow.id)) {
+                it.source.lldp.size() == 1
+                verifyEquals(it.source.lldp.first(), lldpData)
+            }
+        }
+
+        cleanup: "Turn off devices prop, remove connected devices, remove flow"
+        flowHelper.deleteFlow(flow.id)
+        database.removeConnectedDevices(sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialProps)
+    }
+
```

Comment:
I think test for a default flow is missed

Reply:
It is added in #2986